pyqtgraph/dockarea/Dock.py

```diff
@@ -227,6 +226,10 @@ def raiseDock(self):
 
     def close(self):
         """Remove this dock from the DockArea it lives inside."""
+        if self._container is None:
+            warnings.warn(f"Cannot close dock {self} because it is not open.")
```

Comment:
nice touch on the warning! I think you need to add `stacklevel=2` to capture the correct context

```python
warnings.warn(f"Cannot close dock {self} because it is not open.", RuntimeWarning, stacklevel=2)
```

Reply:
good point! fixed in 53fe2b38dd994dfcd7d64f9b5b5848f016bc461b